[Bearypop] iP

Bearypop opened the pull request:
## Beary 🧸
> Pen down your tasks! - Beary
---
### Beary is your personal task-managing chatbot. 
- Easy to use
- Allows you to manage your tasks efficiently
- Simple, easy to understand, text-based UI

To get started:
1. Click [here](https://github.com/nus-cs2103-AY2324S1/ip/pull/49) to download.
2. Launch the application using a suitable version of Java.
3. Add your tasks.
4. Let ***Beary*** manage your tasks for you 🐻
  
Features:
- [x] Managing tasks
- [ ] Having a conversation
- [ ] Doing your tasks
---
***Here's the `run` method:***
```java
public void run() {
        Scanner scanner = new Scanner(System.in);
        Ui.printGreeting(this.name);
        new ListCommand().execute(this.storage, this.tasks);

        while (true) {
            String fullCommand = scanner.nextLine();
            Ui.printLine();

            Parser parser = new Parser();
            Command command = parser.parse(fullCommand);
            if (command == null) {
                continue;
            }
            command.execute(this.storage, this.tasks);
            if (command instanceof ExitCommand) {
                break;
            }
        }
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Bearypop committed:
Added Rename, Greet, Exit feature

Bearypop committed:
Added Echo feature

Bearypop committed:
Added Add and List feature

Bearypop committed:
Add Mark as Done feature

Bearypop committed:
Add ToDos, Events, Deadlines feature

Bearypop committed:
Create test files

Bearypop committed:
Add exception handling

Bearypop committed:
Add delete feature

Bearypop committed:
Add file write and read methods

Bearypop committed:
Add Save feature

Bearypop committed:
Merge branch 'branch-level-7'

Bearypop committed:
Add Date feature

Bearypop committed:
Merge branch 'branch-level-8'

Bearypop committed:
Create new classes for refactoring

Bearypop committed:
Refactor code to make it more OOP

Bearypop committed:
Merge branch 'branch-moreOOP'

Bearypop committed:
Add packages

Bearypop committed:
Setup Gradle

Bearypop committed:
Add JUnit tests for Parser and Storage

Bearypop committed:
Add JavaDoc comments

Bearypop committed:
Add Find feature

Bearypop committed:
Merge branch 'A-JavaDoc'

Bearypop committed:
Merge branch 'Level-9'

Bearypop committed:
Add support for CheckStyle

Bearypop committed:
Add starting files for JavaFX

Bearypop committed:
Add simple GUI

Bearypop committed:
Merge branch 'branch-level-10'

Bearypop committed:
Add a 1 second pause before exiting, when the user types "bye"

LimJH2002 commented on src/main/java/duke/commands/FindCommand.java:
```diff
@@ -0,0 +1,42 @@
+     * @param storage
+     * @param tasks
+     */
+    public void execute(Storage storage, ArrayList<Task> tasks) {
```
Great job for using plural for lists!

LimJH2002 commented on src/main/java/duke/tasks/Task.java:
```diff
@@ -0,0 +1,37 @@
+        this.done = false;
+    }
+
+    public void markDone() {
```
Great job in using verbs for your method names!

LimJH2002 commented on src/main/java/duke/utils/Ui.java:
I like how your method names are concise!

LimJH2002 commented on src/main/java/duke/utils/Storage.java:
I like how your code is being nested nicely!

Bearypop replied:
Thanks for the comments!

jingyu987 commented on src/main/java/duke/commands/ListCommand.java:
```diff
@@ -0,0 +1,23 @@
+package duke.commands;
```
I like how you separated your files into multiple packages instead of one big package. Good job!

Bearypop replied:
Thanks!

jingyu987 commented on src/main/java/duke/utils/Parser.java:
```diff
@@ -0,0 +1,82 @@
+            case "find":
+                return new FindCommand(fullCommand);
+
+            case "delete":
+                try {
+                    int taskNumber = Integer.parseInt(words[1]);
+                    return new DeleteCommand(taskNumber);
+                } catch (IndexOutOfBoundsException e) {
+                    System.out.println("Missing task number");
+                    Ui.printLine();
+                    return null;
+                }
```
You can consider making a separate function to test the validity of commands, will be helpful when you want to add more user error handling.

Bearypop replied:
Okay

AryanG01 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,114 @@
+
+public class Duke {
+    private String name;
+    private ArrayList<Task> tasks;
```
Just a suggestion, you might want to abstract this into a class called TaskList

Bearypop replied:
Okay

AryanG01 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,114 @@
+
+    public void run() {
+        Scanner scanner = new Scanner(System.in);
+        Ui.printGreeting(this.name);
```
Good job on giving it a personal touch for the user

AryanG01 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,114 @@
+        try {
+            this.tasks = storage.readFile();
+        } catch (FileNotFoundException e) {
+            this.tasks = new ArrayList<>();
```
I like how you added error handling here if the file cannot be found

AryanG01 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,114 @@
+     * @param input
+     * @return A string representing duke's response
+     */
+    public String getResponse(String input) {
```
Any reason u decided to have this method here instead of in a command class?

Bearypop replied:
This method is not related to commands, rather it generates Duke's message to be displayed on the frontend.

AryanG01 commented on src/main/java/duke/commands/AddCommand.java:
```diff
@@ -0,0 +1,94 @@
+package duke.commands;
```
Good use of OOP principles by abstracting out all the commands into separate classes

Bearypop replied:
Thanks!

AryanG01 commented on src/main/java/duke/commands/AddCommand.java:
```diff
@@ -0,0 +1,94 @@
+            String description;
+            Task newTask = new Task("");
+            switch (type) {
+                case "todo":
```
Any reason why u didn't add a time option for this like the other event types

Bearypop replied:
todo tasks don't have dates attached to them.

AryanG01 commented on src/main/java/duke/tasks/Deadline.java:
```diff
@@ -0,0 +1,30 @@
+        this.byDate = this.by.format(byFormatter);
+    }
+
+    public Deadline(String description, String by, Boolean other) {
```
A suggestion - You could rename this variable to follow the naming guidelines like the other variables you have named

Bearypop replied:
Isn't the variable in camel case?

Bearypop replied:
Do you mean the variable by?

AryanG01 commented on src/main/resources/images/DaUser.png:
Nice pictures that you are using!

Bearypop committed:
Add assertions

Bearypop committed:
Merge pull request #1 from Bearypop/branch-A-Assertions

Add assertions

Bearypop committed:
Merge branch 'master' of https://github.com/Bearypop/ip

Bearypop committed:
Improve code quality by deleting unused code

Bearypop committed:
Merge pull request #2 from Bearypop/branch-A-CodeQuality

Improve code quality by deleting unused code

Bearypop committed:
Merge branch 'master' of https://github.com/Bearypop/ip

Bearypop committed:
Add workflow file for Github Actions to set up Continuous Integration

Bearypop committed:
Add help command

When the user uses the app for the first time, he or she will not know
what the available commands are.

Adding a new command called "help" will allow the user to view all the
available commands.

Let's create a new command class for the help command, which implements
the command interface, as this will allow us to practice good OOP
principles. We should also add support for parsing this new command and
prompt the user to use this command in the event where they type in an
invalid command.

Bearypop committed:
Change images and touch up GUI

Bearypop committed:
Remove extra newlines in response message

Bearypop committed:
Add product screenshot

Bearypop committed:
Updated README

Bearypop committed:
Change greeting message

Bearypop committed:
Update README

Bearypop committed:
Update screenshot

Bearypop committed:
Refactor code

Bearypop committed:
Add javadoc comments and test cases

Bearypop committed:
Add more javadoc comments

Bearypop committed:
Add additional javadoc comments

Bearypop committed:
Update build.gradle to support linux

Bearypop committed:
Bug fix
